File: app/src/main/java/com/example/comiaseokt/api/ProductoApi.java
package com.example.comiaseokt.api;

import com.example.comiaseokt.entity.GenericResponse;
import com.example.comiaseokt.entity.service.Producto;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ProductoApi {
    String base = "api/producto";

    @GET(base)
    Call<GenericResponse<List<Producto>>> listarProductosRecomendados();

    @GET(base + "/{idC}")
    Call<GenericResponse<List<Producto>>> listarProductosPorCategoria(@Path("idC") int idC);
}

File: app/src/main/java/com/example/comiaseokt/api/ProductoApi.kt
package com.example.comiaseokt.api

import com.example.comiaseokt.activity.GenericResponse
import com.example.comiaseokt.entity.Producto
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path


interface ProductoApi {
    @GET(base)
    fun listarPlatillosRecomendados(): Call<GenericResponse<List<Producto?>?>?>?

    @GET(base + "/{idC}")
    fun listarPlatillosPorCategoria(@Path("idC") idC: Int): Call<GenericResponse<List<Producto?>?>?>?

    companion object {
        const val base = "api/platillo"
    }
}
